backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/graphicpromotion/internal/BpkGraphicPromoImpl.kt

```diff
@@ -0,0 +1,271 @@
+import net.skyscanner.backpack.compose.graphicpromotion.VerticalAlignment
+import net.skyscanner.backpack.compose.overlay.BpkOverlay
+import net.skyscanner.backpack.compose.overlay.BpkOverlayType
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@Composable
+internal fun BpkGraphicPromoImpl(
+    headline: String,
+    image: @Composable BoxScope.() -> Unit,
+    modifier: Modifier = Modifier,
+    kicker: String? = null,
+    subHeadline: String? = null,
+    overlayType: BpkOverlayType? = null,
+    variant: Variant = Variant.OnDark,
+    verticalAlignment: VerticalAlignment = VerticalAlignment.Top,
+    sponsor: Sponsor? = null,
+    sponsorLogo: @Composable () -> Unit? = {},
+    tapAction: () -> Unit = {},
+) {
+    val roundedCornerShape = RoundedCornerShape(BpkBorderRadius.Md)
+    val contentDescription = listOfNotNull(kicker, headline, subHeadline, sponsor?.accessibilityLabel)
+        .joinToString(separator = ", ")
+
+    val interactionSource = remember { MutableInteractionSource() }
+    val isPressed by interactionSource.collectIsPressedAsState()
+
+    Box(
+        modifier = modifier
+            .aspectRatio(RATIO_PORTRAIT)
+            .clip(roundedCornerShape)
+            .clickable(
+                interactionSource = interactionSource,
+                indication = null,
+                onClick = tapAction,
+            )
+            .semantics(mergeDescendants = true) {
+                role = Role.Button
+                this.contentDescription = contentDescription
+            },
+    ) {
+        overlayType?.let {
+            BpkOverlay(
+                modifier = Modifier.fillMaxSize(),
+                overlayType = it,
+                foregroundContent = {
+                    ForegroundContent(
+                        headline = headline,
+                        kicker = kicker,
+                        subHeadline = subHeadline,
+                        variant = variant,
+                        verticalAlignment = verticalAlignment,
+                        sponsor = sponsor,
+                        sponsorLogo = sponsorLogo,
+                    )
+                },
+            ) {
+                InteractiveBackground(image, isPressed)
+            }
+        } ?: run {
+            InteractiveBackground(image, isPressed)
+            ForegroundContent(
+                headline = headline,
+                kicker = kicker,
+                subHeadline = subHeadline,
+                variant = variant,
+                verticalAlignment = verticalAlignment,
+                sponsor = sponsor,
+                sponsorLogo = sponsorLogo,
+            )
+        }
+    }
+}
+
+@Composable
+private fun InteractiveBackground(
+    image: @Composable BoxScope.() -> Unit,
```

Comment:
All Composables need to have a modifier, including composables like this. Also, lambdas with content is usually the last parameter and it's named just `content` in most of the cases.